File: test_data/code/java/1854-maximum-population-year.java
class Solution {
    public int maximumPopulation(int[][] logs) {
        int[] count = new int[101]; // Initialize array with 101 elements
        for (int[] log : logs) {
            for (int year = log[0]; year < log[1]; year++) {
                count[year - 1950]++;
            }
        }
        int year = -1;
        int maxPopulation = 0;
        for (int i = 0; i < count.length; i++) {
            if (count[i] > maxPopulation) {
                maxPopulation = count[i];
                year = i + 1950; // Convert index to actual year
            }
        }
        return year;
    }
}

File: test_data/code/python/1854-maximum-population-year.py
class Solution:
    def maximumPopulation(self, logs: List[List[int]]) -> int:
        births = [(log[0], i) for i, log in enumerate(logs)]
        deaths = [(log[1], i) for i, log in enumerate(logs)]
        births.sort()
        deaths.sort()
        
        population = {}
        alive = set()
        for death, i in deaths:
            population[death] = population.get(death, 0) - 1
            alive.discard(i)
        for birth, i in births:
            population[birth] = population.get(birth, 0) + 1
            alive.add(i)
            
        max_pop = 0
        earliest_year = 0
        for year, count in population.items():
            if count > max_pop:
                max_pop = count
                earliest_year = year
        return earliest_year